### PR TITLE
chore(query): reduce-functions-code-expand

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

 reduce-functions-code-expand

continue #16961 

## Tests

- [x] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/databendlabs/databend/17114)

### CODE REVIEW

**Comment by sundy-li** [comment]
image

After `32s`, there is only on crate left:  common-functions.

We should split this crate into multi crates.

**Review by Xuanwo — APPROVED** [review]
Thank you for this!

File: src/query/expression/src/register_vectorize.rs [hunk 1]
```diff
@@ -0,0 +1,577 @@
...
+        + Copy
+        + Send
+        + Sync,
+) -> impl Fn(Value<I1>, Value<I2>, Value<I3>, Value<I4>, &mut EvalContext) -> Value<O> + Copy + Send + Sync
+{
+    move |arg1, arg2, arg3, arg4, ctx| {
+        let generics = ctx.generics.to_vec();
+        let process_rows = if arg1.as_scalar().is_some()
+            && arg2.as_scalar().is_some()
+            && arg3.as_scalar().is_some()
```
Comment (b41sh) [inline src/query/expression/src/register_vectorize.rs:126]: seems missing `arg4` here?

**Review by b41sh — APPROVED** [review]

**Comment by sundy-li** [comment]
Perf tests are not related to this pr.

**Pull request merged**